pkg/plugin/admissioncontrollers/webhook/conf/am_conf.go

```diff
@@ -0,0 +1,477 @@
+	acc.dumpConfigurationInternal()
+}
+
+func (acc *AdmissionControllerConf) dumpConfigurationInternal() {
+	log.Logger().Info("Loaded admission controller configuration",
+		zap.String("namespace", acc.namespace),
+		zap.String("kubeConfig", acc.kubeConfig),
+		zap.String("policyGroup", acc.policyGroup),
+		zap.String("amServiceName", acc.amServiceName),
+		zap.String("schedulerServiceAddress", acc.schedulerServiceAddress),
+		zap.Strings("processNamespaces", regexpsString(acc.processNamespaces)),
+		zap.Strings("bypassNamespaces", regexpsString(acc.bypassNamespaces)),
+		zap.Strings("labelNamespaces", regexpsString(acc.labelNamespaces)),
+		zap.Strings("noLabelNamespaces", regexpsString(acc.noLabelNamespaces)),
+		zap.Bool("bypassAuth", acc.bypassAuth),
+		zap.Bool("trustControllers", acc.trustControllers),
+		zap.Strings("systemUsers", regexpsString(acc.systemUsers)),
+		zap.Strings("externalUsers", regexpsString(acc.externalUsers)),
+		zap.Strings("externalGroups", regexpsString(acc.externalGroups)))
+}
+
+func regexpsString(regexes []*regexp.Regexp) []string {
+	result := make([]string, 0)
+	for _, regex := range regexes {
+		result = append(result, regex.String())
+	}
+	return result
+}
+
+func parseConfigRegexps(config map[string]string, key string, defaultValue string, env string) []*regexp.Regexp {
+	value := parseConfigString(config, key, defaultValue, env)
+	result, err := parseRegexes(value)
+	if err != nil {
+		log.Logger().Error(fmt.Sprintf("Unable to parse regex values '%s' for configuration '%s', using default value '%s'",
+			value, key, defaultValue), zap.Error(err))
+		result, err = parseRegexes(defaultValue)
+		if err != nil {
+			log.Logger().Fatal("BUG: can't parse default regex pattern", zap.Error(err))
+		}
+	}
+	return result
+}
+
+func parseConfigBool(config map[string]string, key string, defaultValue bool, env string) bool {
+	value := parseConfigString(config, key, fmt.Sprintf("%t", defaultValue), env)
+	result, err := strconv.ParseBool(value)
+	if err != nil {
+		log.Logger().Error(fmt.Sprintf("Unable to parse bool value '%s' for configuration '%s', using default value '%t'",
+			value, key, defaultValue), zap.Error(err))
+		result = defaultValue
+	}
+	return result
+}
+
+func parseConfigInt(config map[string]string, key string, defaultValue int, env string) int {
+	value := parseConfigString(config, key, fmt.Sprintf("%d", defaultValue), env)
+	result, err := strconv.ParseInt(value, 10, 31)
+	if err != nil {
+		log.Logger().Error(fmt.Sprintf("Unable to parse int value '%s' for configuration '%s', using default value '%d'",
+			value, key, defaultValue), zap.Error(err))
+		return defaultValue
+	}
+	return int(result)
+}
+
+func parseConfigString(config map[string]string, key string, defaultValue string, env string) string {
+	result := defaultValue
+	var envValue, confValue string
+	var envOk, confOk bool
+
+	if env != "" {
+		envValue, envOk = os.LookupEnv(env)
+		if envOk {
+			result = envValue
```

Comment:
Just an idea - we might want to display a message with `Warn()` that configmaps are preferred over env variables and suggest migration. If the plan is to completely move away from it, then we can be loud about it.

Reply:
This is temporary code. Once this merges, Helm will be updated to only set the configmap entries and the env card will go away completely. That wasn't done here as it would break intermediate builds.

Reply:
I see, then I'm ok with this.